metrics/cloudwatch/cloudwatch.go

```diff
@@ -133,11 +144,41 @@ func (cw *CloudWatch) Send() error {
 		}
 	}
 
-	_, err := cw.svc.PutMetricData(&cloudwatch.PutMetricDataInput{
-		Namespace:  aws.String(cw.namespace),
-		MetricData: datums,
-	})
-	return err
+	var tokens = make(chan struct{}, cw.numConcurrentRequests)
+	var errors = make(chan error)
+	var n int
+
+	for len(datums) > 0 {
```

Comment:
If you step through my code, you'll see it's pretty much equivalent to what you're proposing.